pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,234 @@
+
+The **game server** will produce the following events: when players join the server, pick up items in-game, uses the chat, hit one another, and eventually disconnect. It will be implemented to simulate players at random intervals is joining the server, picking up some items, etc, and eventually disconnecting, to provide a sense of realism. 
+
+The backend **processor** will be consuming these events to process them. In this series, I will not do anything particular with the data other than it will simply save the received events directly to a database.
+
+I will not get into the specifics of the stack for this system yet since it does not affect the writing of the API documents for the two applications.
+
+# Designing the APIs with AsyncAPI
+When starting designing the application APIs I always use the [design first principle](https://apisyouwonthate.com/blog/api-design-first-vs-code-first), even when we are talking about internal systems.
+## The game server
+I always start with the basics, and first define all the different channels for which the **game server** should produce events over. 
+
+```yaml
+asyncapi: 2.0.0
+info: 
+  title: "Game server"
+  version: "0.0.1"
+channels: 
+  game/server/{serverId}/events/player/{playerId}/item/{itemId}/pickup:
+    description: Channel used when a player picks up an item in-game
+  game/server/{serverId}/events/player/{playerId}/connect:
+    description: Channel used when a player joins (connect to) the game server
+  game/server/{serverId}/events/player/{playerId}/disconnect:
+    description: Channel used when a player leaves (disconnects from) the game server
+  game/server/{serverId}/events/player/{playerId}/chat: 
+    description: Channel used when a player writes something in chat
+  game/server/{serverId}/events/player/{playerId}/hit: 
+    description: Channel used when a player hit another player in-game
+```
+
+AsyncAPI channels have a different meaning based on the underlying setup, for brokers such as [Apache Kafka](https://kafka.apache.org/), this is referred to as `topics`. 
+
+However, regardless of the underlying setup, channels must be defined as a [RFC 6570 URI template](https://www.asyncapi.com/docs/specifications/2.0.0#channelsObject). 
+
+The way I like to structure my channels is to utilize parameters to separate the action from information about the event, so it describes, on what server the event was performed `{serverId}`, by what player `{playerId}` and in case of **pickup**, what item `{itemId}` gets picked up. For the last part of the channel I describe what the event was, **pickup**, **connect**, **disconnect**, etc.
```

Comment:
```suggestion
The way I like to structure my channels is to utilize parameters to separate the action from information about the event, so it describes, on what server the event was performed `{serverId}`, by what player `{playerId}` and in case of **pickup**, what item `{itemId}` gets picked up. For the last part of the channel, I describe what event it was, **pickup**, **connect**, **disconnect**, etc.
```